File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String,Integer> board = new HashMap<>();
        
        for (String p : participant) {
            board.put(p, board.getOrDefault(p,0) + 1);
        }
        
        for (String c : completion) {
            board.put(c, board.get(c) - 1);
        }
        
        for (String k : board.keySet()) {
            if (board.get(k) != 0) {
                answer = k;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ""
    par = {}
    
    for i in participant :
        if i in par :
            par[i] += 1
        else :
            par[i] = 1
    
    for i in completion :
        par[i] -= 1
    
    for i in participant :
        if par[i] != 0 :
            answer = i
    return answer